bots/tests-scan

```diff
@@ -214,6 +236,14 @@ def tests_invoke(priority, name, revision, ref, context, base, repo, bots_ref, o
         repo=pipes.quote(repo),
     )
 
+def queue_test(priority, name, revision, ref, context, base, repo, bots_ref, queue, options):
+    command = tests_invoke(priority, name, revision, ref, context, base, repo, bots_ref, options)
+    if command:
+        if priority > MAX_PRIORITY:
+            priority = MAX_PRIORITY
+        msg = amqp.Message(body=command, priority=priority)
```

Comment:
Let's please wrap this into JSON for keeping it extensible. Also can't hurt to add the task type ("test"), PR number and SHA as properties.

Reply:
ugh i was going to do that wasn't i, i'll be sure to add it, sorry for forgetting it again

Reply:
as the pr number isn't immediately available at that point, i just added the sha, name, and ref. the name always contains the pull number and the ref if not empty also does.